src/main/host/syscall_handler.c

```diff
@@ -179,8 +173,8 @@ SysCallReturn syscallhandler_make_syscall(SysCallHandler* sys,
 
     // Lazily initialize memoryManager on the first syscall. It needs a thread
     // in the syscall state for its initialization.
-    if (!sys->memoryManager) {
-        sys->memoryManager = memorymanager_new(sys->thread);
+    if (!process_getMemoryManager(sys->process)) {
+        process_setMemoryManager(sys->process, memorymanager_new(sys->thread));
```

Comment:
So, iiuc, in the syscall handling code, `process_getMemoryManager()` should always return non-NULL. So should the NULL checks above (with fallback to NATIVE) be an assertion error instead?

Reply:
Ah, at the last minute I was trying to make it easy to disable the MM completely for experimentation by just commenting out the line where it gets initialized. Since we don't have immediate plans for that I'll go ahead and make these asserts rather than falling back.

Reply:
Ah, I mean e.g. in mman.c we can drop the fallbacks to NATIVE. Here we're lazily initializing it; I'll add better comments.